### PR TITLE
Add getL1BatchDetails, tidy docs a wee bit, update Types, include Custom Bridge updates DOC-288

### PR DESCRIPTION
**Pull request by kailash-manasarovar**
- Added `getL1BatchDetails`
- Some wordage fixes: TS vs JS, function name vs explanation, alpha list
- Copy/paste `Types` code from source ... to be documented properly shortly in DOC-276
- Added Custom Bridge updates from DOC-288 also

### CODE REVIEW

**Comment by uF4No** [comment]
I won't change the name to Typescript. It's written in Typescript but you can use it with plain JS and, everyone know JS but not all JS developers know Typescript.

**Comment by kailash-manasarovar** [comment]
> I won't change the name to Typescript. It's written in Typescript but you can use it with plain JS and, everyone know JS but not all JS developers know Typescript.

It's so imprecise. Are you sure?

File: docs/dev/developer-guides/bridging/bridging-asset.md [hunk 1]
```diff
@@ -173,7 +171,35 @@ Users must call the `withdraw` method on the L2 bridge contract, which will trig
...
 
 ::: warning
-
 On the testnet environment, we automatically finalize all withdrawals, i.e., for every withdrawal, we will take care of it by making an L1 transaction that proves the inclusion for each message.
-
 :::
+
+## Custom bridges
+
+Implement your own custom bridge by creating a new `L1ERC20BridgeFactory` and using it to deploy a custom bridge by passing the `CONTRACTS_DIAMOND_PROXY_ADDR` and an `allowlist` for the contract signer to the `deploy` function.
```
Comment (Deniallugo) [inline docs/dev/developer-guides/bridging/bridging-asset.md:179]: I'm not sure it's true. 
`L1ERC20Bridge` and `Factory` is our implementation of the ERC20 bridge. 

And everything you described below is about our bridge implementation, not the custom one, is it on purpose?

Comment (kailash-manasarovar) [reply docs/dev/developer-guides/bridging/bridging-asset.md:179]: I wrote this from looking at the example code provided which is where I got the snippet I put below the text. The link to the example was in the ticket here: https://github.com/matter-labs/zksync-2-dev/pull/1633/files#diff-143fd8bd5920c0327f92ee25b6bd377322d122d3d45424a2e5dc92cc60a2096e please go ahead and correct or if it's a bigger job, we can have a quick call?

Comment (kailash-manasarovar) [reply docs/dev/developer-guides/bridging/bridging-asset.md:179]: got it, thank you

Comment (kailash-manasarovar) [reply docs/dev/developer-guides/bridging/bridging-asset.md]: DONE, please check.

File: docs/dev/developer-guides/bridging/bridging-asset.md [hunk 2]
```diff
@@ -173,7 +171,16 @@ Users must call the `withdraw` method on the L2 bridge contract, which will trig
...
+## Custom bridges on L1 and L2
+
+To build a custom bridge, create a regular Solidity contract which extends the correct interface for the layer.
+
+- L1: [IL1Bridge.sol](https://github.com/matter-labs/era-contracts/blob/main/ethereum/contracts/bridge/interfaces/IL1Bridge.sol)
+- L2: [L2ERC20Bridge.sol](https://github.com/matter-labs/era-contracts/blob/main/zksync/contracts/bridge/L2ERC20Bridge.sol)
+
+The interfaces provide access to the zkSync Era SDK deposit and withdraw implementations.
+
+For more info, check out an example [custom bridge implementation](https://github.com/matter-labs/zksync-2-dev/blob/main/core/tests/ts-integration/tests/custom-erc20-bridge.test.ts).
```
Comment (Deniallugo) [inline docs/dev/developer-guides/bridging/bridging-asset.md:186]: 1. It's the private repo
2. It's just a test of how we did deploy it

Probably the best way is to point to our implementation of the bridges. 
https://github.com/matter-labs/era-contracts/blob/main/ethereum/contracts/bridge/L1ERC20Bridge.sol


https://github.com/matter-labs/era-contracts/blob/main/zksync/contracts/bridge/L2ERC20Bridge.sol

**Comment by kailash-manasarovar** [comment]
> I won't change the name to Typescript. It's written in Typescript but you can use it with plain JS and, everyone know JS but not all JS developers know Typescript.

OK, updated.

**Review by Deniallugo — APPROVED** [review]

**Pull request merged**